### PR TITLE
Nircam move f444 w grism

### PR DESCRIPTION
**Pull request by Johannes-Sahlmann**
Addressing https://jira.stsci.edu/browse/JWSTSIAF-129.

### CODE REVIEW

File: generate/generate_nircam.py [hunk 1]
```diff
@@ -165,6 +167,9 @@
                 aperture.V3Ref += v3_offset
             elif dependency_type == 'dhspil_wedge':
                 aperture.V3Ref += 43.
+            # elif dependency_type == 'grism_f444w':
```
Comment (shanosborne) [inline generate/generate_nircam.py]: Delete commented out code?

File: generate/generate_nircam.py [hunk 2]
```diff
@@ -318,100 +342,126 @@
 aperture_collection = pysiaf.ApertureCollection(aperture_dict)
 
 emulate_delivery = True
-emulate_delivery = False
+# emulate_delivery = False
```
Comment (shanosborne) [inline generate/generate_nircam.py]: Delete commented out code?

File: generate/generate_nircam.py [hunk 3]
```diff
@@ -318,100 +342,126 @@
...
     pre_delivery_dir = os.path.join(JWST_DELIVERY_DATA_ROOT, instrument)
     if not os.path.isdir(pre_delivery_dir):
         os.makedirs(pre_delivery_dir)
 
     # write the SIAF files to disk
-    filenames = pysiaf.iando.write.write_jwst_siaf(aperture_collection, basepath=pre_delivery_dir, file_format=['xml', 'xlsx'])
+    filenames = pysiaf.iando.write.write_jwst_siaf(aperture_collection, basepath=pre_delivery_dir, file_format=['xml'], label='jwstsiaf-129')
 
-    pre_delivery_siaf = pysiaf.Siaf(instrument, basepath=pre_delivery_dir)
+    # 1/0
```
Comment (shanosborne) [inline generate/generate_nircam.py]: Delete commented out code?

File: generate/generate_nircam.py [hunk 4]
```diff
@@ -318,100 +342,126 @@
...
+    # pre_delivery_siaf = pysiaf.Siaf(instrument, basepath=pre_delivery_dir)
+    pre_delivery_siaf = pysiaf.Siaf(instrument, filename=filenames[0])
 
     # compare new SIAF with PRD version
-    ref_siaf = pysiaf.Siaf(instrument)
-    compare.compare_siaf(pre_delivery_siaf, reference_siaf_input=ref_siaf, fractional_tolerance=1e-6, report_dir=pre_delivery_dir, tags={'reference': pysiaf.JWST_PRD_VERSION, 'comparison': 'pre_delivery'})
+    # ref_siaf = pysiaf.Siaf(instrument)
```
Comment (shanosborne) [inline generate/generate_nircam.py]: Maybe want to try and find a way to move the checks on lines 360, 362, and 366 somewhere more appropriate like a separate function? Or if that isn't needed, just delete them?

File: generate/generate_nircam.py [hunk 5]
```diff
@@ -318,100 +342,126 @@
...
-# roundtrip_table = compare.compare_transformation_roundtrip(new_siaf, reference_siaf_input=ref_siaf, selected_aperture_name=[s for s in aperture_name_list if 'FULL' in s], report_dir=test_dir, make_plot=False)
-
-1/0
-
-# illustrate fix of inverse NIRCam coefficients
-if 0:
-    import pylab as pl
-
-    pl.close('all')
+    if 0:
```
Comment (shanosborne) [inline generate/generate_nircam.py]: Could this check go in a separate function to keep the code base clean?

File: generate/generate_nircam.py [hunk 6]
```diff
@@ -318,100 +342,126 @@
...
+        pl.xticks(np.arange(len(roundtrip_table)), roundtrip_table['AperName'], rotation='vertical')
+        pl.margins(0.2)
+        pl.subplots_adjust(bottom=0.15)
+        pl.show()
+
+        roundtrip_table[roundtrip_table['AperName']=='NRCA5_FULL'].pprint()
+
+
+
+    if 0:
```
Comment (shanosborne) [inline generate/generate_nircam.py]: Could this check go in a separate function?

File: generate/generate_nircam.py [hunk 7]
```diff
@@ -318,100 +342,126 @@
...
+    # roundtrip_table = compare.compare_transformation_roundtrip(new_siaf, reference_siaf_input=ref_siaf, selected_aperture_name=[s for s in aperture_name_list if 'FULL' in s], report_dir=test_dir, make_plot=False)
 
+    1/0
 
-if 0:
-    ref_siaf = pysiaf.Siaf(instrument, basepath=os.path.join(pysiaf.constants._DATA_ROOT, 'JWST','PRDOPSSOC-G-012','SIAFXML/SIAFXML'))
-    new_siaf = pysiaf.Siaf(instrument)
-    compare.compare_siaf(new_siaf, reference_siaf_input=ref_siaf, fractional_tolerance=1e-6)
+    # illustrate fix of inverse NIRCam coefficients
+    if 0:
```
Comment (shanosborne) [inline generate/generate_nircam.py]: Could this check go in a separate function?

File: generate/generate_nircam.py [hunk 8]
```diff
@@ -318,100 +342,126 @@
...
+    # compare.compare_siaf(new_siaf, reference_siaf_input=ref_siaf, fractional_tolerance=1e-4, ignore_attributes=ignore_attributes, report_dir=test_dir)
+    # compare.compare_siaf(new_siaf, reference_siaf_input=ref_siaf, fractional_tolerance=1e-6)
 
-    roundtrip_table[roundtrip_table['AperName']=='NRCA5_FULL'].pprint()
 
+    roundtrip_table = compare.compare_transformation_roundtrip(new_siaf, reference_siaf_input=ref_siaf, report_dir=test_dir)
+    # roundtrip_table = compare.compare_transformation_roundtrip(new_siaf, reference_siaf_input=ref_siaf, selected_aperture_name='NRCA5_FULL'.split(), report_dir=test_dir, make_plot=True)
+    # roundtrip_table = compare.compare_transformation_roundtrip(new_siaf, reference_siaf_input=ref_siaf, selected_aperture_name=[s for s in aperture_name_list if 'FULL' in s], report_dir=test_dir, make_plot=False)
 
+    1/0
```
Comment (shanosborne) [inline generate/generate_nircam.py]: There's a leftover `1/0` here

File: generate/generate_nircam.py [hunk 9]
```diff
@@ -318,100 +342,126 @@
...
-    # pl.plot(roundtrip_table['siaf1_dx_rms'], 'ko--', label='fixed dx_mean')
-    # pl.plot(roundtrip_table['siaf1_dy_rms'], 'go--', label='fixed dy_mean')
-    # pl.title('RMS absolute difference')
+    new_siaf = pysiaf.Siaf(instrument, filenames[0])
 
-    pl.xticks(np.arange(len(roundtrip_table)), roundtrip_table['AperName'], rotation='vertical')
-    pl.margins(0.2)
-    pl.subplots_adjust(bottom=0.15)
-    pl.show()
+    # compare.compare_siaf(new_siaf, reference_siaf_input=ref_siaf, fractional_tolerance=1e-1, selected_aperture_name=master_aperture_names)#['NRCA3_FULL_OSS', 'NRCA1_FULL_OSS']) # 'NRCA4_SUB160', 'NRCA4_FULL', 'NRCA3_SUB160', 'NRCA3_FULL', 'NRCA5_SUB400P', 'NRCB5_SUB400P',
```
Comment (shanosborne) [inline generate/generate_nircam.py]: Could all the compare checks (lines 413-416 and 421-422, 426-427) be moved somewhere else (like a separate function) or just deleted?

File: generate/generate_nircam.py (hunk 1 above)
Comment (Johannes-Sahlmann) [reply generate/generate_nircam.py]: done

File: generate/generate_nircam.py (hunk 2 above)
Comment (Johannes-Sahlmann) [reply generate/generate_nircam.py]: done

File: generate/generate_nircam.py (hunk 3 above)
Comment (Johannes-Sahlmann) [reply generate/generate_nircam.py]: done. 
I cleaned this part up in the next commit. I will resolve the applicable comments.

**Pull request merged**